dexter/src/main/java/com/karumi/dexter/DexterInstance.java

```diff
@@ -64,15 +67,39 @@
   }
 
   /**
-   * Checks the state of a specific permission reporting it when ready to the listener
+   * Checks the state of a specific permission reporting it when ready to the listener.
+   * Listener.onPermissionGranted and Listener.onPermissionDenied are called on the same thread
+   * that fired the permission request.
+   *
+   * @param listener The class that will be reported when the state of the permission is ready
+   * @param permission One of the values found in {@link android.Manifest.permission}
+   */
+  void checkPermissionOnSameThread(PermissionListener listener, String permission) {
+    checkSinglePermission(listener, permission, ThreadSpecFactory.makeSameThreadSpec());
+  }
```

Comment:
Instead of create this new method can't we parametrize the one below? That method has default visibility and we can change the method signature without adding breaking changes.